test/e2e/incluster/leaderelection_test.go

```diff
@@ -0,0 +1,96 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build e2e
+// +build e2e
+
+package incluster
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func testLeaderElection(namespace string) {
+	// This test is solely a check on the fact that we have set up leader
```

Comment:
for reviewers, if you have any thoughts about how to do this, couldn't figure out a good way to observe which instance sets status on a resource

Reply:
No good ideas of the top of my head, I thought maybe the managed fields metadata would help but looks like it just sets the manager to "contour", so fine with this for now.

Reply:
Was just reading https://kubernetes.io/docs/reference/using-api/server-side-apply/#managers, looks like _maybe_ if we explicitly set the user agent based on the pod name in [here](https://github.com/projectcontour/contour/blob/011fe4023796769e8762587f7e7136c432549720/internal/k8s/clients.go#L41) then the manager would show up as that, would have to do some experimenting though. Wouldn't consider it a blocker for this PR.

Reply:
hm yeah confirmed you would have to have an update to the httpproxy or ingress resource you want to observe a different manager on, unless we make some significant changes to the status updater bits to make them rewrite status on all resources when leader election changes

not too bad, but if we're doing it just for a test, probably not worth it, not sure if theres other significant value we get